packages/web3-eth/test/unit/rpc_methods/estimate_gas.test.ts

```diff
@@ -0,0 +1,69 @@
+/*
+This file is part of web3.js.
+
+web3.js is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+web3.js is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+// web3.js is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// web3.js is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Lesser General Public License for more details.
+
+// You should have received a copy of the GNU Lesser General Public License
+// along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+// */
+import { Web3RequestManager } from 'web3-core';
+import { validator } from 'web3-validator';
+
+import { estimateGas } from '../../../src/rpc_methods';
+import { testData } from './fixtures/estimate_gas';
+
+jest.mock('web3-validator');
```

Comment:
I prefer not to only mock the I/O or compute intensive tasks. In this case if we have right input/output then the validator will not be a problem for the tests. What do you think?

Reply:
I mocked the validator to be able to spy on the `.validate` method to ensure it was being called with the expected data. Since we have explicit tests for `web3-validator`, I don't think we need to be concerned that the validation passes. Instead, I opted for verifying that we're validating the expected data with the expected validation rules (e.g. `blockNumberOrTag`, `string`, `hex`, etc.)